src/consumer/message.rs

```diff
@@ -0,0 +1,48 @@
+use std::marker::PhantomData;
+
+use crate::{
+    consumer::data::MessageData,
+    message::proto::{MessageIdData, MessageMetadata},
+    DeserializeMessage, Payload,
+};
+
+/// a message received by a consumer
+///
+/// it is generic over the type it can be deserialized to
+pub struct Message<T> {
+    /// origin topic of the message
+    pub topic: String,
+    /// contains the message's data and other metadata
+    pub payload: Payload,
+    /// contains the message's id and batch size data
+    pub message_id: MessageData,
+    pub(crate) _phantom: PhantomData<T>,
```

Comment:
```suggestion
    pub(super) _phantom: PhantomData<T>,
```

Narrow the scope as possible. Previously it was private.